### DISCUSSION TITLE
Unpoly 3.6.0 released

### DISCUSSION THREAD

**Discussion by triskweline** [unpoly/unpoly#564] [Announcements]
### Targeting fragments

- Unpoly-specific pseudo selectors like [`:main`](https://unpoly.com/main) or [`:layer`](https://unpoly.com/layer) can now be used in a compound target, e.g. `:main .child`.
- Targeting [`:main`](https://unpoly.com/main) will no longer [match in the region of the interaction origin](https://unpoly.com(/targeting-fragments#resolving-ambiguous-selectors)). It will now always use the first matching selector in [`up.fragment.config.mainTargets`](https://unpoly.com/up.fragment.config#config.mainTargets).
- Fix a bug where following a navigation item outside a [main](https://unpoly.com/main) element would focus the `<body>` instead of the main element.

### Performance improvements

- Unpoy now uses the native [`:has()`](https://unpoly.com/has) selector [where available](https://developer.mozilla.org/en-US/docs/Web/CSS/:has). Unpoly's polyfill for [`:has()`](https://unpoly.com/has) will remain included for the time being. It will be removed as Firefox' [`:has()`](https://unpoly.com/has) support has reached the majority of users (available on Nightly now). 
- Improve performance of many element lookups, by finding elements via CSS selectors (vs. filtering lists with JavaScript).

### Support for [structured data markup](https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data)

- Structured data in `script[type="application/ld+json"]` elements is considered a meta tag that will be [updated with history changes](https://unpoly.com/updating-history#history-state).
- `script[type="application/ld+json"]` elements in are now preserved in new fragments with `up.fragment.config.runScripts = false`.

### Bugfixes and minor improvements

- CSP nonces [embedded into attribute callbacks](https://unpoly.com/csp#nonceable-attributes) now work with [`Content-Security-Policy-Report-Only`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only).
- When the `X-Up-Validate` header value exceeds 2048 characters, it is now set to `:unknown`.
  This is to prevent web infrastructure from rejecting an overly long request line with an `413 Entity Too Large` error.
- Fix a bug where `up:assets:changed` would be emitted for every response when configuring `up.fragment.config.runScripts = false`.
- `up.form.isSubmittable()` now returns `false` for forms with a cross-origin URL in their `[action]` attribute.
- `up.util.contains()` now works on `NodeList` objects.
- You can now [configure](https://unpoly.comup.script.config.scriptSelectors) which elements are removed by `up.fragment.config.runScripts = false`.

**Comment by alnuaimi94** [comment]
I'm using vite tool and I installed unpoly using npm,
the problem is the size of bundle js file when i run build 
it became around 173KB.

**Reply by hpvd** [reply]
@alnuaimi94 maybe you are interested in this discussion: https://github.com/unpoly/unpoly/discussions/547

**Comment by louisgavalda** [comment]
Hi from France,

Thank you for maintaining such a delicious piece of software! ;-)

Best regards.

**Reply by triskweline** [reply]
> Thank you for maintaining such a delicious piece of software! ;-)

This made my day :heart:
